ozaria/site/components/play/PageIntroLevel/index.vue

```diff
@@ -69,163 +61,137 @@
+            this.currentContentId = ''
+            return
           }
-        },
-        fetchNextLevel: async function () {
+          this.currentContentId = content.contentId[this.language]
+        } else {
+          this.currentContentId = content.contentId
+        }
+        // reload key is concatenated into child component's keys
+        // so that we can trigger their re-render with the same content id by incrementing reloadKey
+        // this is needed when `replay` is clicked on victory modal
+        this.reloadKey[content.type] = this.reloadKey[content.type] || 0
+        this.reloadKey[content.type]++
+      },
+      setIntroLevelComplete: async function () {
+        if (!me.isSessionless()) { // not saving progress/session for teachers
           try {
-            const currentLevel = this.campaignData.levels[this.introLevelData.original]
-            const nextLevelData = getNextLevelForLevel(currentLevel) || {}
-            const nextLevelOriginal = nextLevelData.original
-            this.nextLevelStage = nextLevelData.nextLevelStage
-            if (nextLevelOriginal) {
-              this.nextLevel = await api.levels.getByOriginal(nextLevelOriginal)
-            }
+            this.introLevelSession.state.complete = true
+            await api.levelSessions.update(this.introLevelSession)
           } catch (err) {
-            console.error('Error in fetching next level', err)
+            console.error('Error in saving intro level session', err)
             // TODO handle_error_ozaria
-            noty({ text: 'Error in fetching next level', type: 'error', timeout: 2000 })
-          }
-        },
-        fetchNextLevelLink: function () {
-          if (this.nextLevel.slug && this.nextLevel.type) {
-            const nextLevelOptions = {
-              courseId: this.courseId,
-              courseInstanceId: this.courseInstanceId,
-              campaignId: this.campaignId,
-              codeLanguage: this.codeLanguage,
-              nextLevelStage: this.nextLevelStage
-            }
-            return getNextLevelLink(this.nextLevel, nextLevelOptions)
-          } else {
-            console.log('no next level found')
-            return `/ozaria/play/${this.campaignData.slug}`
+            return noty({ text: 'Error in saving intro level session', type: 'error', timeout: 2000 })
           }
         }
       }
-    )
+    }
   })
 </script>
 
 <template>
   <div v-if="dataLoaded">
     <interactives-component
       v-if="currentContent.type == 'interactive'"
-      :key="currentContentId"
+      :key="currentContentId + `${reloadKey[currentContent.type]}`"
       :interactive-id-or-slug="currentContentId"
       :code-language="language"
       @completed="onContentCompleted"
     />
     <cinematics-component
       v-else-if="currentContent.type == 'cinematic'"
-      :key="currentContentId"
+      :key="currentContentId + `${reloadKey[currentContent.type]}`"
```

Comment:
Could this be written like:

`${currentContentId}${reloadKey[currentContent.type]}`

Also does the reload number need to be tied to an object? Could it just be set statically on the data and incremented?

Reply:
I wanted to keep it differentiated for different contents like interactives/cinematics/cutscenes, technically using a single value would also work, but then any change would impact the keys for all the child components.

Reply:
That's true. But only one content type is shown at a time. This is fine.

Reply:
yeah thats true, but keeping it different would be cleaner to understand and less brittle.